pulpcore/content/handler.py

```diff
@@ -1125,13 +1130,27 @@ async def finalize():
                 await original_finalize()
 
         downloader = remote.get_downloader(
-            remote_artifact=remote_artifact, headers_ready_callback=handle_response_headers
+            remote_artifact=remote_artifact,
+            headers_ready_callback=handle_response_headers,
         )
         original_handle_data = downloader.handle_data
         downloader.handle_data = handle_data
         original_finalize = downloader.finalize
         downloader.finalize = finalize
-        download_result = await downloader.run()
+        try:
+            download_result = await downloader.run()
+        except DigestValidationError:
+            # Cant recover from wrong data already sent.
+            # We should close the connection without sending an EOF in the response
```

Comment:
Don't wash the message up with words like "should"... We decided we **do** it at this point. Also maybe we don't need this comment at all. The whole text is covered by the RuntimeError in the same codeblock.

```suggestion
            # We close the connection without sending an EOF in the response.
```